web/src/left_sidebar_navigation_area.js

```diff
@@ -143,6 +143,34 @@ export function highlight_inbox_view() {
     }, 0);
 }
 
+function handle_home_view_order(home_view) {
+    // Remove class and tabindex from current home view
+    const $current_home_view = $(".selected-home-view");
+    $current_home_view.removeAttr("tabindex");
+    $current_home_view.removeClass("selected-home-view");
+
+    const $all_messages_rows = $(".top_left_all_messages");
+    const $recent_views_rows = $(".top_left_recent_view");
+    const $inbox_rows = $(".top_left_inbox");
+
+    // Add the class and tabindex to the matching home view
+    if (home_view === settings_config.web_home_view_values.all_messages.code) {
+        $all_messages_rows.addClass("selected-home-view");
+        $all_messages_rows.find("a").attr("tabindex", 0);
+    } else if (home_view === settings_config.web_home_view_values.recent_topics.code) {
+        $recent_views_rows.addClass("selected-home-view");
+        $recent_views_rows.find("a").attr("tabindex", 0);
+    } else {
+        // Inbox is home view.
+        $inbox_rows.addClass("selected-home-view");
+        $inbox_rows.find("a").attr("tabindex", 0);
+    }
+    // This is probably not the best way to do this,
+    // but we need to make sure there's an unread
+    // count on the new chosen home view.
+    unread_ui.update_unread_counts();
```

Comment:
This is probably fine, given that it's a rare event. But the more precise thing we could do is `left_sidebar_navigation_area.update_dom_with_unread_counts` ... but that requires passing the result of `const res = unread.get_counts()`.